### PR TITLE
Handle buffers as literal atoms when used as a Filter.

### LINKED ISSUE #128: Filter on binary data
I can write the buffer and retrieve it again using decode:false but I cannot figure out how to filter on the value.

```js
const buf = Buffer.from('a468c3a669', 'hex');

// Throws Can't convert to RegExp String from unknown type
{
   value: buf
}

// Returns zero rows instead of throwing
{
  value: [
    buf
  ]
}

// Using binary string also returns zero rows
{
   value: buf.toString('binary')
}
```

#### Environment details

  - `@google-cloud/bigtable` version: 0.13.1

### PR DESCRIPTION
**Pull request by kolodny**
Fixes #128 

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: src/filter.js [hunk 1]
```diff
@@ -803,6 +808,17 @@ Filter.prototype.toProto = function() {
  * ];
  *
  * //-
+ * // Or you can provide a Buffer or an array of Buffers if you wish to match
+ * // against specfic binary value(s).
+ * //-
+ * var userInputedFaces = [Buffer.from('.|.'), Buffer.from(':-)')]
```
Comment (stephenplusplus) [inline src/filter.js]: Trailing `;`.

Comment (kolodny) [reply src/filter.js]: Fixed

File: src/filter.js [hunk 2]
```diff
@@ -111,6 +112,10 @@ Filter.convertToRegExpString = function(regex) {
     return regex.toString();
   }
 
+  if (Buffer.isBuffer(regex)) {
+    return escapeStringRegexp(regex.toString());
```
Comment (moander) [inline src/filter.js]: This will create an utf-8 string. Using `.toString('binary')` will preserve the actual bytes

File: test/filter.js [hunk 3]
```diff
@@ -65,6 +65,13 @@ describe('Bigtable/Filter', function() {
...
 
+    it('should convert an Array of buffers to a single string', function() {
+      var faces = [Buffer.from('.|.'), Buffer.from('=|=')];
+      var str = Filter.convertToRegExpString(faces);
+
+      assert.strictEqual(str, '(\\.\\|\\.|=\\|=)');
+    });
+
     it('should not do anything to a string', function() {
       var str1 = 'hello';
```
Comment (moander) [inline test/filter.js]: Should test some special characters to make sure strings are not escaped

File: src/filter.js (hunk 2 above)
Comment (moander) [reply src/filter.js]:
```js
Buffer.from(Buffer.from('e6','hex').toString())
// <Buffer ef bf bd>

Buffer.from(Buffer.from('e6','hex').toString('binary'))
// <Buffer c3 a6>

Buffer.from(Buffer.from('e6','hex').toString('binary'), 'binary')
// <Buffer e6>
```

Comment (moander) [reply src/filter.js]: Maybe use `\xFF` escaping for non-ASCII bytes?

**Pull request merged**